ddtrace/internal/datastreams/kafka.py

```diff
@@ -11,15 +11,42 @@
 
 
 INT_TYPES = (int,)
+MESSAGE_ARG_POSITION = 1
+KEY_ARG_POSITION = 2
+KEY_KWARG_NAME = "key"
+
+
+def _calculate_byte_size(data):
+    if isinstance(data, str):
+        # We encode here to handle non-ascii characters
+        return len(data.encode("utf-8"))
```

Comment:
This will [raise an exception](https://docs.python.org/3/library/stdtypes.html#str.encode) on non-ascii characters. For the purpose of calculating payload size, maybe `errors='replace'` is appropriate.

Reply:
thank you!

Reply:
thanks! fixed